### PR TITLE
Issue #272: Add command groups.

### LINKED ISSUE #272: Group similar commands together
With the growing list of commands in Bee, it'd be nice to group similar commands together when viewing a list of them all.

For example:
```
 STATE
  state-get             Get the value of a Backdrop state.
  sg, sget

  state-set             Set the value of an existing Backdrop state.
  ss, sset

 USER
  user-login            Display a login link for a given user.
  uli

  user-password         Reset the login password for a given user.
  upw, upwd

  users                 List all user accounts.
  uls, user-list
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop-contrib/bee/issues/272

### CODE REVIEW

File: commands/projects.bee.inc [hunk 1]
```diff
@@ -12,6 +12,7 @@ function projects_bee_command() {
     'projects' => array(
       'description' => bt('Display information about available projects (modules, themes, layouts).'),
       'callback' => 'projects_bee_callback',
+      'group' => 'projects',
```
Comment (yorkshire-pudding) [inline commands/projects.bee.inc:15]:
```suggestion
      'group' => 'information',
```

File: commands/php.bee.inc [hunk 2]
```diff
@@ -12,6 +12,7 @@ function php_bee_command() {
     'eval' => array(
       'description' => bt('Evaluate (run/execute) arbitrary PHP code after bootstrapping Backdrop.'),
       'callback' => 'eval_bee_callback',
+      'group' => 'miscellaneous',
```
Comment (yorkshire-pudding) [inline commands/php.bee.inc]:
```suggestion
      'group' => 'advanced',
```

File: commands/install.bee.inc [hunk 3]
```diff
@@ -12,6 +12,7 @@ function install_bee_command() {
     'install' => array(
       'description' => bt('Install Backdrop and setup a new site.'),
       'callback' => 'install_bee_callback',
+      'group' => 'core',
```
Comment (yorkshire-pudding) [inline commands/install.bee.inc:15]:
```suggestion
      'group' => 'installation',
```

File: commands/download.bee.inc [hunk 4]
```diff
@@ -31,6 +32,7 @@ function download_bee_command() {
     'download-core' => array(
       'description' => bt('Download Backdrop core.'),
       'callback' => 'download_core_bee_callback',
+      'group' => 'core',
```
Comment (yorkshire-pudding) [inline commands/download.bee.inc:35]:
```suggestion
      'group' => 'installation',
```

File: commands/dblog.bee.inc [hunk 5]
```diff
@@ -12,6 +12,7 @@ function dblog_bee_command() {
     'log' => array(
       'description' => bt('Show database log messages.'),
       'callback' => 'log_bee_callback',
+      'group' => 'miscellaneous',
```
Comment (yorkshire-pudding) [inline commands/dblog.bee.inc]:
```suggestion
      'group' => 'information',
```

File: commands/php.bee.inc [hunk 6]
```diff
@@ -26,6 +27,7 @@ function php_bee_command() {
     'php-script' => array(
       'description' => bt('Execute an arbitrary PHP file after bootstrapping Backdrop.'),
       'callback' => 'script_bee_callback',
+      'group' => 'miscellaneous',
```
Comment (yorkshire-pudding) [inline commands/php.bee.inc]:
```suggestion
      'group' => 'advanced',
```

File: commands/state.bee.inc [hunk 7]
```diff
@@ -12,6 +12,7 @@ function state_bee_command() {
     'state-get' => array(
       'description' => bt('Get the value of a Backdrop state.'),
       'callback' => 'state_get_bee_callback',
+      'group' => 'state',
```
Comment (yorkshire-pudding) [inline commands/state.bee.inc:15]:
```suggestion
      'group' => 'miscellaneous',
```

File: commands/state.bee.inc [hunk 8]
```diff
@@ -24,6 +25,7 @@ function state_bee_command() {
     'state-set' => array(
       'description' => bt('Set the value of an existing Backdrop state.'),
       'callback' => 'state_set_bee_callback',
+      'group' => 'state',
```
Comment (yorkshire-pudding) [inline commands/state.bee.inc:28]:
```suggestion
      'group' => 'miscellaneous',
```

File: commands/status.bee.inc [hunk 9]
```diff
@@ -12,6 +12,7 @@ function status_bee_command() {
     'status' => array(
       'description' => bt('Provides an overview of the current Backdrop installation/site.'),
       'callback' => 'status_bee_callback',
+      'group' => 'miscellaneous',
```
Comment (yorkshire-pudding) [inline commands/status.bee.inc]:
```suggestion
      'group' => 'information',
```

File: commands/update.bee.inc [hunk 10]
```diff
@@ -12,6 +12,7 @@ function update_bee_command() {
     'update-db' => array(
       'description' => bt('Show, and optionally apply, all pending database updates.'),
       'callback' => 'update_db_bee_callback',
+      'group' => 'core',
```
Comment (yorkshire-pudding) [inline commands/update.bee.inc]: This is not limited to core - works for module updates too.
```suggestion
      'group' => 'update',
```

File: commands/version.bee.inc [hunk 11]
```diff
@@ -12,6 +12,7 @@ function version_bee_command() {
     'version' => array(
       'description' => bt('Display the current version of Bee.'),
       'callback' => 'version_bee_callback',
+      'group' => 'miscellaneous',
```
Comment (yorkshire-pudding) [inline commands/version.bee.inc]:
```suggestion
      'group' => 'information',
```

**Review by yorkshire-pudding — CHANGES_REQUESTED** [review]
The functionality is good, but I think the wiki groupings are better and so it would be best to align these to the categories we have there.

**Comment by ghost** [comment]
@yorkshire-pudding Let's discuss grouping in the main issue before changing them here...

**Review by yorkshire-pudding — APPROVED** [review]

**Pull request merged**